### DISCUSSION TITLE
Get Stripe Token from Payment Sheets

### DISCUSSION THREAD

**Discussion by davidoort** [flutter-stripe/flutter_stripe#1124] [Q&A]
Hi! I'm trying to find an easy way to get a stripe token from a payment sheet so that I can pass that token to a 3rd party which then actually charges the customer.

I know I can use a credit card form for this but it is a lot less convenient for the user than a payment sheet since that offers different payment methods and the ability to save payment details for future purchases. However, I haven't found a way to extract the payment method token after doing initPaymentSheet(customFlow: true) and presentPaymentSheet(). 

If this is not possible, can someone confirm that the output of the [createApplePayToken](https://pub.dev/documentation/flutter_stripe/latest/flutter_stripe/Stripe/createApplePayToken.html) method is the same as the [createToken](https://pub.dev/documentation/flutter_stripe/latest/flutter_stripe/Stripe/createToken.html) method, and clarify why there is no such method for Google Pay?

Thank you!

**Comment by davidoort** [comment]
Is there maybe something similar to onPaymentResult callback that the Pay plugin provides for the Payment Sheets?

**Comment by jamesdixon** [comment]
@davidoort did you ever figure this out? I can't even see how to create a token with card details. We're on the old charges API so I can't use payment intents

**Comment by davidoort** [comment]
You'll have to use the Credit Card form of the Pay plugin. I haven't figured out how to do it through Payment Sheets
